R/pkg/R/SQLContext.R

```diff
@@ -17,6 +17,34 @@
 
 # SQLcontext.R: SQLContext-driven functions
 
+#' Temporary function to reroute old S3 Method call to new
+#' We need to check the class of x to ensure it is SQLContext before dispatching
+dispatchFunc <- function(newFuncSig, x, ...) {
+  funcName <- as.character(sys.call(sys.parent())[[1]])
+  f <- get0(paste0(funcName, ".default"))
+  # Strip sqlContext from list of parameters and then pass the rest along.
+  # In the following, if '&' is used instead of '&&', it warns about
+  # "the condition has length > 1 and only the first element will be used"
+  if (class(x) == "jobj" &&
+      grepl("org.apache.spark.sql.SQLContext", capture.output(show(x)))) {
+    .Deprecated(newFuncSig, old = paste0(funcName, "(sqlContext...)"))
+    f(...)
+  } else {
+    f(x, ...)
+  }
+}
```

Comment:
Thought: dispatchFunc strips sqlContext and calls into .default methods. Is it better for dispatchFunc to add sqlContext if it does not exist and calls into .default methods which accepts sqlContext as its first argument? This allows a use case that user can pass in different sqlContext, say a sqlContext and a hiveContext. It seems that sqlContext and hiveContext can co-exist?